internal/service/s3/bucket.go

```diff
@@ -81,18 +81,21 @@ func ResourceBucket() *schema.Resource {
 				Optional:      true,
 				ConflictsWith: []string{"grant"},
 				ValidateFunc:  validation.StringInSlice(BucketCannedACL_Values(), false),
+				Deprecated:    "Use the aws_s3_bucket_acl resource instead",
 			},
 
 			"grant": {
 				Type:          schema.TypeSet,
 				Optional:      true,
 				Set:           grantHash,
 				ConflictsWith: []string{"acl"},
+				Deprecated:    "Use the aws_s3_bucket_acl resource instead",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"id": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:       schema.TypeString,
+							Optional:   true,
+							Deprecated: "Use the aws_s3_bucket_acl resource instead",
```

Comment:
I'm not sure if we need `Deprecated` on all fields, or just the top level

Reply:
Hmm yeah i'm not actually sure but i could it see it being a bit unnecessary to have it on the nested fields. i'll remove 👍